src/init.cpp

```diff
@@ -723,6 +723,19 @@ bool AppInitServers(boost::thread_group& threadGroup)
     return true;
 }
 
+[[noreturn]] static void new_handler_terminate()
+{
+    // Rather than throwing std::bad-alloc if allocation fails, terminate
+    // immediately to (try to) avoid chain corruption.
+    // Since LogPrintf may itself allocate memory, set the handler directly
+    // to terminate first.
+    std::set_new_handler(std::terminate);
```

Comment:
Take @kallewoof's [suggestion](https://github.com/bitcoin/bitcoin/pull/9856#discussion_r103093216) to do
```
fputs(stderr, "Error: Out of memory. Terminating.\n");
```
here.